src/transformers/models/superpoint/modeling_superpoint.py

```diff
@@ -298,3 +298,41 @@ def forward(
             last_hidden_state=last_hidden_state,
             hidden_states=encoder_outputs.hidden_states,
         )
+
+
```

Comment:
SuperPointModelForInterestPointDescription is more or less the exact same as SuperPointModel.
I just use it for my own understanding of the transformers library since I'm taking inspiration on other implementations (like ConvNextV2). So either SuperPointModel or SuperPointModelForInterestPointDescription might be deleted later

Reply:
In transformers, the class `XXXModel` does not contain the head on top and the classes `XXXForYYY` add different head models. `XXX` represents the model name and `YYY` represents a task.

Reply:
So the proposed implementation is correct regarding this convention ? Because I can't really tell what would be considered as a head in SuperPoint. Or shoud `SuperPointModel` only contain the `encoder` and the `SuperPointForInterestPointDescription` contain the `keypoint_decoder` and `descriptor_decoder` ? And then in this case `SuperPointModel.forward()` should output a `BackboneOutput` similar to ConvNextV2 ?

Reply:
Looking at the paper and code structure, the SuperPoint is a traditional CNN based model, and I couldn't identify what could be identified as the `head`.

The encoder is simply a CNN which returns features (`last_hidden`). The 2 decoders `keypoint_decoder` and `descriptor_decoder`, return keypoints+scores and descriptors, respectively. So, I think the structure that best fits this case is:

* The `SuperPointModel` should only contain the `encoder`, a `SuperPointEncoder` object. Then, the `SuperPointModel.forward()` should output a `BaseModelOutputWithNoAttention` object.
* The `SuperPointForInterestPointDescription`should contain the `superpoint`, a `SuperPointModel` object, and both `keypoint_decoder` and `descriptor_decoder`. Then, the `SuperPointForInterestPointDescriptor.forward()` should output a `ImagePointDescriptionOutput` object.

@amyeroberts , what do you think of this structure? Makes sense?

Reply:
If I understood correctly, I need to remove every mentions of SuperPointForInterestPointDescription and only keep SuperPointModel that can be instantiated by AutoModelForInterestPointDescription ? [this commit](https://github.com/huggingface/transformers/pull/25786/commits/de606ee114f3f1fd19a5f5cddc53fe0c966a79cd) reflects these changes, let me know if I misunderstood something